bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/DataLayerProperties.java

```diff
@@ -0,0 +1,158 @@
+package com.adobe.cq.wcm.core.components.models;
+
+import org.apache.sling.api.resource.Resource;
+import org.osgi.annotation.versioning.ConsumerType;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import java.util.Map;
+
+/**
+ * A base interface to be extended by components that need to enable dataLayer population.
+ */
+@ConsumerType
+public interface DataLayerProperties {
+
+    /**
+     * Indicates if the component will populate the dataLayer
+     *
+     * @return {@code true} if the model will populate the dataLayer
+     */
+    @JsonIgnore
+    default boolean isDataLayerEnabled() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Method used to return a resource's asset (ex {@link Image})
+     *
+     * @return {@code Resource} of the asset
+     */
+    @JsonIgnore
+    default Resource getAssetResource() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Method used to return a model's ID
+     *
+     * @return string ID
+     */
+    @JsonIgnore
+    default String getDataLayerId() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Method used to return a model's type
+     *
+     * @return type
+     */
+    @JsonIgnore
+    default String getDataLayerType() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Method used to return a model's link (ex {@link Image})
+     *
+     * @return src
+     */
+    @JsonIgnore
+    default String getDataLayerTitle() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Method used to return a model's text where it applies (ex {@link Text})
+     *
+     * @return text
+     */
+    @JsonIgnore
+    default String getDataLayerText() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Method used to return a model's tags where it applies (ex {@link Page})
+     *
+     * @return tags array
+     */
+    @JsonIgnore
+    default String[] getDataLayerTags() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Method used to return a model's link where a URL is present
+     *
+     * @return link URL
+     */
+    @JsonIgnore
+    default String getDataLayerLinkUrl() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Method used to return a page's template path
+     *
+     * @return JCR template path
+     */
+    @JsonIgnore
+    default String getDataLayerTemplatePath() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Get a model's language where it applies (ex {@link Page}
+     *
+     * @return language
+     */
+    @JsonIgnore
+    default String getDataLayerLanguage() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Get an {@link Carousel} or {@link Tabs} active item IDs
+     *
+     * @return item ID
+     */
+    @JsonIgnore
+    default String[] getDataLayerShownItems() {
```

Comment:
I'd rephrase it: Returns the shown items of the Carousel, Accordion or Tabs used in the data layer